node/nodeTransactions.go

```diff
@@ -31,70 +29,79 @@ func (n *Node) GetTransaction(txHash string) (*transaction.ApiTransactionResult,
 		return nil, err
 	}
 
+	tx, err := n.optionallyGetTransactionFromPool(hash)
+	if err != nil {
+		return nil, err
+	}
+	if tx != nil {
+		return tx, nil
+	}
+
 	if n.historyRepository.IsEnabled() {
 		return n.getFullHistoryTransaction(hash)
 	}
 
-	return n.getTransaction(hash)
+	return n.getTransactionFromCurrentEpochStorage(hash)
 }
 
-func (n *Node) getTransaction(hash []byte) (*transaction.ApiTransactionResult, error) {
+func (n *Node) optionallyGetTransactionFromPool(hash []byte) (*transaction.ApiTransactionResult, error) {
 	txObj, txType, found := n.getTxObjFromDataPool(hash)
-	if found {
-		return n.castObjToTransaction(txObj, txType)
+	if !found {
+		return nil, nil
 	}
 
-	txBytes, txType, found := n.getTxBytesFromStorage(hash)
-	if found {
-		return n.unmarshalTransaction(txBytes, txType)
+	tx, err := n.castObjToTransaction(txObj, txType)
+	if err != nil {
+		return nil, err
 	}
 
-	return nil, fmt.Errorf("transaction not found")
+	tx.Status = (&transaction.StatusComputer{
+		SourceShard:      tx.SourceShard,
+		DestinationShard: tx.DestinationShard,
+		Receiver:         tx.Tx.GetRcvAddr(),
+		TransactionData:  tx.Data,
+		SelfShard:        n.shardCoordinator.SelfId(),
+	}).ComputeStatusWhenInPool()
+
+	return tx, nil
 }
 
 func (n *Node) getFullHistoryTransaction(hash []byte) (*transaction.ApiTransactionResult, error) {
-	// get transaction from pool
-	txObj, txType, found := n.getTxObjFromDataPool(hash)
-	if found {
-		// transaction is in pool return information directly because in history storer
-		// there are no additional information about the transaction
-		return n.castObjToTransaction(txObj, txType)
-	}
-
 	miniblockMetadata, err := n.historyRepository.GetMiniblockMetadataByTxHash(hash)
 	if err != nil {
-		// transaction is not in history storer
-		return nil, err
+		return nil, ErrTransactionNotFound
 	}
 
 	txBytes, txType, found := n.getTxBytesFromStorageByEpoch(hash, miniblockMetadata.Epoch)
 	if !found {
-		// this should never happen because transaction was found in history storer should be also found in transaction
-		// storer
-		log.Warn("node transaction: cannot find transaction in storage")
-		return putHistoryFieldsInTransaction(nil, miniblockMetadata), nil
+		log.Warn("getFullHistoryTransaction(): unexpected condition, cannot find transaction in storage")
+		return nil, ErrCannotRetrieveTransaction
 	}
 
 	tx, err := n.unmarshalTransaction(txBytes, txType)
 	if err != nil {
-		// this should never happen
-		log.Warn("node transaction: cannot unmarshal transaction", "error", err.Error())
-		return putHistoryFieldsInTransaction(nil, miniblockMetadata), nil
+		log.Warn("getFullHistoryTransaction(): unexpected condition, cannot unmarshal transaction")
+		return nil, ErrCannotRetrieveTransaction
 	}
 
-	// merge data about transaction from history storer and transaction storer
-	tx = putHistoryFieldsInTransaction(tx, miniblockMetadata)
+	putHistoryFieldsInTransaction(tx, miniblockMetadata)
```

Comment:
maybe you can add this method inside structure  `ApiTransactionResult`
`
func (ats *ApiTransactionResult) PutHistoryFieldsInTransaction(miniblockMetadata *fullHistory.MiniblockMetadata)` just a suggestion

Reply:
That is a good suggestion, indeed, to encapsulate this logic in `ApiTransactionResult`. The issue is, though, that ` ApiTransactionResult` is defined in `transaction` package, while `miniblockMetadata` in `fullHistory` package - and that would require an additional interface - we can do in the future.